### PR TITLE
Added JSDOC for initDismissibles.ts  for #9495

### PR DESCRIPTION
**Pull request by Yekasumah**
I added JSDOC to the updateDismissibles and initDismissibles functions in the initDismissibles.ts files. I was able to create a new branch that didn't have any of my previously committed changes, thanks to your guidelines. As for the tests, somehow it's still failing and I'm still working on it to fix the error causing it to fail.

### CODE REVIEW

**Comment by Yekasumah** [comment]
Hi @lb-, I have added the unit tests in the recent commit. Please review.

File: client/src/includes/initDismissibles.ts [hunk 1]
```diff
@@ -1,36 +1,55 @@
-import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
-
-export function updateDismissibles(data: Record<string, boolean>) {
-  return fetch(WAGTAIL_CONFIG.ADMIN_URLS.DISMISSIBLES, {
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig'
```
Comment (lb-) [inline client/src/includes/initDismissibles.ts:1]: Check you have run the formatter I think there should be a line break here 

`npm run format`

File: client/src/includes/initDismissibles.ts [hunk 2]
```diff
@@ -1,36 +1,55 @@
...
+/**
+ * Updates the server, using a PATCH request when the toggle is clicked on a dismissible
+ * element initialized by initDismissibles
```
Comment (lb-) [inline client/src/includes/initDismissibles.ts:4]:
```suggestion
 * element initialised by initDismissibles
```

**Review by lb- — CHANGES_REQUESTED** [review]
A few recommendations for changes. This is looking really great so far. 

Unit tests look pretty solid also - well done.

**Comment by lb-** [comment]
@Yekasumah - checking in on this - are you still able to work through the review feedback?

**Comment by Yekasumah** [comment]
@lb- 
Yes I have been able to do it. I’ll push the changes to the pr now.

**Review by lb- — APPROVED** [review]
@Yekasumah thanks so much for this, I will push up a bit of a refinement to the unit tests that resolve some of the comments but just wanted to let you know for context.

In general, these unit tests could be much simpler and I have just revised to the bare minimum. I wanted to thank you for being thorough but for now just a small basic suite is all we need.

Also, we do not need to write unit tests in TypeScript but I realise that was not clarified anywhere - sorry fo the confusion with that but thanks for taking the time to get it working anyway.

I really appreciate you contributing this clean up and thanks again.

File: client/src/includes/initDismissibles.ts [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
-
-export function updateDismissibles(data: Record<string, boolean>) {
-  return fetch(WAGTAIL_CONFIG.ADMIN_URLS.DISMISSIBLES, {
+/**
+ * Updates the server, using a PATCH request when the toggle is clicked on a dismissible
+ * element initialised by initDismissibles
+ *
+ * @param data {Record<string, boolean>} The dismissible represented as a key-value pair
+ * of its id and its new state: whether it is dismissed
+ * @param wagtailConfig {WagTailConfig}
```
[the remark this answers is not in the document]
Comment (Yekasumah) [reply client/src/includes/initDismissibles.ts]: Okay , thanks.

File: client/src/includes/initDismissibles.ts [hunk 4]
```diff
@@ -11,26 +28,28 @@ export function updateDismissibles(data: Record<string, boolean>) {
     mode: 'same-origin',
   });
 }
-
```
[the remark this answers is not in the document]
Comment (Yekasumah) [reply client/src/includes/initDismissibles.ts:14]: Thanks

File: client/src/includes/initDismissibles.test.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
+import { initDismissibles, updateDismissibles } from './initDismissibles';
+const mockFetch = jest.fn(() =>
```
[the remark this answers is not in the document]
Comment (Yekasumah) [reply client/src/includes/initDismissibles.test.ts]: Oh yeah, you’re right

**Comment by Yekasumah** [comment]
Hi @lb- , Thank you very much. I really appreciate your patience with this issue. Also thanks for giving me the opportunity to work on this, I learnt a lot. Prior to now I had not much knowledge about writing unit tests. Now with your corrections and guidance, I’ll be able to improve and get better. Thank you very much, I really enjoyed working on this.

**Comment by lb-** [comment]
Thanks for the reply @Yekasumah - you have done well here and it's hard to know where to draw the line sometimes with unit tests. 

I hope it was fun! 

Please let me know if you are happy with just `Yekasumah` as your name in the contributors list or if you want a full name listed. 

Have a great weekend

**Comment by Yekasumah** [comment]
Hi @lb-, Yes it was very much fun. Yes, I'm very much happy with Yekasumah, it's the short form of my name Onyeka Asumah. Thank You.

**Comment by lb-** [comment]
Awesome I will leave the name as is then.

**Pull request merged**